cmake/feature_test.cmake

```diff
@@ -0,0 +1,45 @@
+include(CheckCXXSourceCompiles)
+
+function(BEMAN_FEATURE_TEST FEATURE_TEST_EXP RESULT_VAR)
+    check_cxx_source_compiles(
+        "
+    #if (${FEATURE_TEST_EXP})
+    int main() {}
+    #else
+    #error
+    #endif
+    "
+        ${RESULT_VAR}
+    )
+endfunction()
+
+function(BEMAN_MIN_CXX_VERSION CXX_VERSION)
```

Comment:
I'm ok merging names like this, but keep for future reference, it's good to pick a "module name" and be consistent about prefixing any function names, variables, or anything else that might collide with other CMake entities anywhere in the CMake ecosystem. See here:
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Help/dev/documentation.rst#module-functions-and-macros

No changes necessary right now because I'm pretty sure we'll end up turning this into a packaged module at some point anyway. If it's named `Beman`, then no name changes are needed.

Reply:
This is exemplar, so we shouldn't rush anything. I would rather we get this right than we rush this in.

Right now, we have an internal arg used explicitly by CI and presets named `BEMAN_BUILDSYS_SANITIZER`.

Would it be a good idea to fold everything into this `BEMAN_BUILDSYS` name space?

Or would it be better called something like `BEMAN_FEATURE_TEST_ENFORCE_MIN_CXX`?

Reply:
I think a `BEMAN_` prefix is enough for now. I was just noting the best practice. If we make a function, variable, option, or any other CMake "name" specific to a project, the "name" must also be somehow specific to the project. Everyone that's not me help keep an eye on that because it's a *real* pain to clean up later when someone finally notices that `FetchContent` doesn't work because everyone's using the same `BUILD_INTEGRATION_TESTS` variable or something.